applications/tari_base_node/src/builder.rs

```diff
@@ -495,43 +493,155 @@ where
+                                .await
+                                .map_err(|e| {
+                                    error!(
+                                        target: LOG_TARGET,
+                                        "Problem starting UTXO validation protocols in the Output Manager Service: {}",
+                                        e
+                                    );
+                                    e
+                                });
+                            trace!(target: LOG_TARGET, "Attempting UTXO validation for Unspent Outputs.",);
+                        },
+                    };
+                    loop {
+                        futures::select! {
+                            event = event_stream.select_next_some() => {
+                                match event.unwrap() {
+                                    // Restart the protocol if aborted (due to 'BaseNodeNotSynced')
+                                    OutputManagerEvent::UtxoValidationAborted(s) => {
+                                        trace!(
+                                            target: LOG_TARGET,
+                                            "UTXO validation event 'UtxoValidationAborted' ({}), restarting.", s,
+                                        );
+                                        break;
+                                    },
+                                    // Restart the protocol if failure
+                                    OutputManagerEvent::UtxoValidationFailure(s) => {
+                                        trace!(
+                                            target: LOG_TARGET,
+                                            "UTXO validation event 'UtxoValidationFailure' ({}), restarting.", s,
+                                        );
+                                        break;
+                                    },
+                                    // Exit upon success
+                                    OutputManagerEvent::UtxoValidationSuccess(s) => {
+                                        trace!(
+                                            target: LOG_TARGET,
+                                            "UTXO validation event 'UtxoValidationSuccess' ({}).", s,
+                                        );
+                                        if validation == ValidationTask::InvalidOutputs {
+                                            validation = ValidationTask::UnspentOutputs;
+                                        } else {
+                                            end_protocol = true;
+                                        }
+                                        break;
+                                    },
+                                    // Wait for the next event if timed out (several can be attempted)
+                                    OutputManagerEvent::UtxoValidationTimedOut(s) => {
```

Comment:
Use this to decide when to restart and also check that `s` is the ID of the validation protocol you care about.

Reply:
No, the monitoring loop should not restart with this event, as it is a valid event that can occur in the validation protocol. The code should just wait for the next event.

Reply:
That is because you are ignoring the ID that is contained in `s` and are worried that you will get a timeout event for a different protocol. You are creating your own timeout when a timeout already exists for the protocol.

Reply:
Even if the `futures::select! {` delay is removed, the monitoring loop should still not be restarted if an `OutputManagerEvent::UtxoValidationTimedOut` event occurs, it should just wait for the next event.

The logic is:
- Restart the validation protocol:
  - `OutputManagerEvent::UtxoValidationAborted`
  - `OutputManagerEvent::UtxoValidationFailure`
- Exit (success) if
  - `OutputManagerEvent::UtxoValidationSuccess`
- Wait for the next event:
  - `OutputManagerEvent::UtxoValidationTimedOut`
  - `OutputManagerEvent::Error`
  - _

Reply:
but the UtxoValidationTimedOut event is exactly the same thing as your timeout but its built in? Why ignore it rather than just use it to restart the process?

Reply:
Fixed - event timeout restart added in the correct location